qt/scientific_interfaces/Indirect/Reduction/ISISEnergyTransferData.h

```diff
@@ -0,0 +1,285 @@
+};
+
+class IETGroupingType {
+public:
+  static inline const std::string DEFAULT = "Default";
+  static inline const std::string FILE = "File";
+  static inline const std::string GROUPS = "Groups";
+  static inline const std::string CUSTOM = "Custom";
+  static inline const std::string IPF = "IPF";
+  static inline const std::string ALL = "All";
+  static inline const std::string INDIVIDUAL = "Individual";
+};
+
+class IETGroupingData {
+public:
+  IETGroupingData(const std::string &groupingType = IETGroupingType::DEFAULT, const int &nGroups = 0,
+                  const std::string &groupingMapFile = "", const std::string &customGroups = "")
+      : m_groupingType(groupingType), m_nGroups(nGroups), m_groupingMapFile(groupingMapFile),
+        m_customGroups(customGroups) {}
+
+  std::string getGroupingType() const { return m_groupingType; }
+  int getNGroups() const { return m_nGroups; }
+  std::string getGroupingMapFile() const { return m_groupingMapFile; }
+  std::string getCustomGroups() const { return m_customGroups; }
+
+private:
+  std::string m_groupingType;
+  int m_nGroups;
+  std::string m_groupingMapFile;
+  std::string m_customGroups;
+};
+
+class IETBackgroundData {
+public:
+  IETBackgroundData(const bool &removeBackground = false, const int &backgroundStart = 0, const int &backgroundEnd = 0)
+      : m_removeBackground(removeBackground), m_backgroundStart(backgroundStart), m_backgroundEnd(backgroundEnd) {}
+
+  bool getRemoveBackground() const { return m_removeBackground; }
+  int getBackgroundStart() const { return m_backgroundStart; }
+  int getBackgroundEnd() const { return m_backgroundEnd; }
+
+private:
+  bool m_removeBackground;
+  int m_backgroundStart;
+  int m_backgroundEnd;
+};
+
+class IETAnalysisData {
+public:
+  IETAnalysisData(const bool &useDetailedBalance = false, const double &detailedBalance = 0.0,
+                  const bool &useScaleFactor = false, const double &scaleFactor = 0.0)
+      : m_useDetailedBalance(useDetailedBalance), m_detailedBalance(detailedBalance), m_useScaleFactor(useScaleFactor),
+        m_scaleFactor(scaleFactor) {}
+
+  bool getUseDetailedBalance() const { return m_useDetailedBalance; }
+  double getDetailedBalance() const { return m_detailedBalance; }
+  bool getUseScaleFactor() const { return m_useScaleFactor; }
+  double getScaleFactor() const { return m_scaleFactor; }
+
+private:
+  bool m_useDetailedBalance;
+  double m_detailedBalance;
+  bool m_useScaleFactor;
+  double m_scaleFactor;
+};
+
+class IETRebinData {
+public:
+  IETRebinData(const bool &doNotRebin = false, const std::string &rebinType = "", const double &rebinLow = 0.0,
+               const double &rebinHigh = 0.0, const double &rebinWidth = 0.0, const std::string &rebinString = "")
+      : m_doNotRebin(doNotRebin), m_rebinType(rebinType), m_rebinLow(rebinLow), m_rebinWidth(rebinWidth),
+        m_rebinHigh(rebinHigh), m_rebinString(rebinString) {}
+
+  bool getDoNotRebin() const { return m_doNotRebin; }
```

Comment:
why is this do not? When looking at the calls to this object, it looks odd to have `IETRebinData(True` if you dont want to rebin the data